Swift Code Challenge/Views/UserDetailsView.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  UserDetailsView.swift
+//  Swift Code Challenge
+//
+//  Created by Luiz Eduardo Mello dos Reis on 30/06/22.
+//
+
+import SwiftUI
+import MessageUI
+
+struct UserDetailsView: View {
+    @State var user: Person
+    var body: some View {
+        VStack{
+            MapView(lat: user.location.coordinates?.latitude, long: user.location.coordinates?.latitude)
+                .frame(height: 200)
+                .ignoresSafeArea(edges: .top)
+            VStack{
+                CircleImage(userImage: user.picture.large ?? "")
+                
+                contentBody(user: user)
+            }
+            .offset(y: -200)
+            .padding(.bottom, -130)
+            
+            Spacer()
+            
+            Text("made by @luizmello.dev - click me!")
+                .onTapGesture {
+                    openInstagram(instagramHandle: "luizmello.dev")
+                }
+                .padding(.vertical)
+        }
+    }
+    func openInstagram(instagramHandle: String) {
+        guard let url = URL(string: "https://instagram.com/\(instagramHandle)")  else { return }
+        if UIApplication.shared.canOpenURL(url) {
+            if #available(iOS 10.0, *) {
+                UIApplication.shared.open(url, options: [:], completionHandler: nil)
+            } else {
+                UIApplication.shared.openURL(url)
+            }
+        }
+    }
+}
+
+struct contentBody: View {
+    @State var result: Result<MFMailComposeResult, Error>? = nil
+    @State var isShowingMailView = false
+    var user: Person
+    
+    var body: some View {
+        VStack(spacing: 5) {
+            Text("\(user.name.first ?? "Luiz") \(user.name.last ?? "Mello")")
+                .bold()
+                .font(.title)
+            HStack() {
+                Text("\(user.location.street?.name ?? "SADSAD"),")
+                    .font(.body)
+                    .foregroundColor(.secondary)
+                Text("\(user.location.city ?? "asdsad") - \(user.location.country ?? "asdad")")
+                    .font(.body)
+                    .bold()
+                    .foregroundColor(.secondary)
+            }
+            .padding(.bottom, 70)
+            
+            Button(action: {
+                let tel = "mailto:"
+                var formattedString = tel + user.email
```

Comment:
No need for this to be a `var`.

```suggestion
                let formattedString = tel + user.email
```